numpy/_core/src/multiarray/convert_datatype.c

```diff
@@ -51,11 +51,22 @@ NPY_NO_EXPORT npy_intp REQUIRED_STR_LEN[] = {0, 3, 5, 10, 10, 20, 20, 20, 20};
 /*
  * Whether or not legacy value-based promotion/casting is used.
  */
-NPY_NO_EXPORT int npy_promotion_state = NPY_USE_LEGACY_PROMOTION;
+
 NPY_NO_EXPORT PyObject *NO_NEP50_WARNING_CTX = NULL;
 NPY_NO_EXPORT PyObject *npy_DTypePromotionError = NULL;
 NPY_NO_EXPORT PyObject *npy_UFuncNoLoopError = NULL;
 
+NPY_NO_EXPORT NPY_TLS int npy_promotion_state = NPY_USE_LEGACY_PROMOTION;
```

Comment:
```suggestion
static NPY_TLS int npy_promotion_state = NPY_USE_LEGACY_PROMOTION;
```
Make it static now that it you have the function accessors.

Reply:
Thanks for pointing this out, it didn't occur to me but of course this is nicer.